datasette/views/special.py

```diff
@@ -140,42 +144,43 @@ async def get(self, request):
                         "name": p.name,
                         "abbr": p.abbr,
                         "description": p.description,
-                        "takes_database": p.takes_database,
-                        "takes_resource": p.takes_resource,
-                        "default": p.default,
+                        "takes_parent": p.takes_parent,
+                        "takes_child": p.takes_child,
                     }
-                    for p in self.ds.permissions.values()
+                    for p in self.ds.actions.values()
                 ],
             },
         )
 
     async def post(self, request):
-        await self.ds.ensure_permissions(request.actor, ["view-instance"])
-        if not await self.ds.permission_allowed(request.actor, "permissions-debug"):
+        if not await self.ds.allowed(action="view-instance", actor=request.actor):
+            raise Forbidden("view-instance")
+        if not await self.ds.allowed(action="permissions-debug", actor=request.actor):
             raise Forbidden("Permission denied")
         vars = await request.post_vars()
         actor = json.loads(vars["actor"])
         permission = vars["permission"]
         resource_1 = vars["resource_1"]
         resource_2 = vars["resource_2"]
-        resource = []
-        if resource_1:
-            resource.append(resource_1)
-        if resource_2:
-            resource.append(resource_2)
-        resource = tuple(resource)
-        if len(resource) == 1:
-            resource = resource[0]
-        result = await self.ds.permission_allowed(
-            actor, permission, resource, default="USE_DEFAULT"
+        # Convert to Resource object
+        if resource_1 and resource_2:
+            resource_obj = TableResource(database=resource_1, table=resource_2)
+            resource_for_response = (resource_1, resource_2)
+        elif resource_1:
+            resource_obj = DatabaseResource(database=resource_1)
+            resource_for_response = resource_1
+        else:
+            resource_obj = InstanceResource()
+            resource_for_response = None
```

Comment:
This looks wrong. We can tell which class we should be using by looking up the action by name in `ds.actions[action_name]` and then accessing its `resource_class` property.